core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -386,7 +387,7 @@ public final void runInContext() {
             }
           } catch (Throwable t) {
             stream.cancel(Status.CANCELLED.withCause(t).withDescription("Failed to read message."));
-            return;
+            Throwables.propagateIfPossible(t);
```

Comment:
This is still going to go nowhere.  It will probably get catch by the Context's executor but there really isn't any code that can handle this exception but the stream.